src/Workspaces/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -181,7 +181,9 @@ public static Task<SyntaxTreeIndex> GetSyntaxTreeIndexAsync(this Document docume
             if (document.Project.TryGetCompilation(out var compilation))
             {
                 // We already have a compilation, so at this point it's fastest to just get a SemanticModel
-                return await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                GC.KeepAlive(compilation);
```

Comment:
Probably doesn't hurt to add a comment that we want to make sure the compilation doesn't get GCed, lest GetSemanticModelAsync() becomes expensive.

Reply:
It looks somewhat obvious and I'm a bit worried about the build servers workload.